torch/compiler/__init__.py

```diff
@@ -149,3 +150,23 @@ def wrap_numpy(fn):
     """
     from torch._dynamo.external_utils import wrap_numpy as wrap
     return wrap(fn)
+
+_is_compiling_flag = False
+
+def is_compiling() -> bool:
+    """
+    Indicates whether a graph is executed as part of torch.compile() or torch.export().
+
+    Note that there are 2 other related flags (we should probably deprecate them eventually):
+      * torch._dynamo.external_utils.is_compiling() https://github.com/pytorch/pytorch/blob/a7f82b7d628eb2b966bc53e593dcf32049b2b10e/torch/_dynamo/external_utils.py#L14
+      * torch._utils.is_compiling() https://github.com/pytorch/pytorch/blob/a7f82b7d628eb2b966bc53e593dcf32049b2b10e/torch/_utils.py#L852
+
+    Example::
+
+        >>> def forward(self, x):
+        >>>     if not torch.compiler.is_compiling():
+        >>>        ...logic that is not needed in a compiled/traced graph...
+        >>>
+        >>>     ...rest of the function...
+    """
+    return _is_compiling_flag
```

Comment:
Given that there are already 2 other flags in the code base, I am not sure about increased complexity of having 3 flags doing the same thing :)

cc @suo, @ezyang, @mlazos

Reply:
The other ones are technically non-public APIs, so I think we should redirect them to the public one and slowly try to extinguish them :P

Reply:
I think you should be able to replace torch._utils.is_compiling() with the torch._dynamo.eternal_utils.is_compiling() right now.

Reply:
> I think you should be able to replace torch._utils.is_compiling() with the torch._dynamo.eternal_utils.is_compiling() right now.

Thanks for reviewing! Yeah, I can do this as part of this diff.